fetch/src/main/java/com/indix/gocd/s3fetch/FetchExecutor.java

```diff
@@ -72,4 +72,7 @@ public AmazonS3Client s3Client(FetchConfig config) {
         return client;
     }
 
+    public FetchConfig getFetchConfig(TaskConfig config, TaskExecutionContext context) {
```

Comment:
It is used in one place - FetchExecutor.java:84.  The reason that was needed is to mock out system environment variables in GoEnvironment (otherwise environment variable on the build server end up in your test and one may get inconsistent results from test to test).